rllib/offline/json_reader.py

```diff
@@ -235,30 +252,76 @@ def _next_file(self) -> FileType:
             path = random.choice(self.files)
         return self._try_open_file(path)
 
+    def _from_json(self, data: str) -> SampleBatchType:
+        if isinstance(data, bytes):  # smart_open S3 doesn't respect "r"
+            data = data.decode("utf-8")
+        json_data = json.loads(data)
 
-def _from_json(batch: str) -> SampleBatchType:
-    if isinstance(batch, bytes):  # smart_open S3 doesn't respect "r"
-        batch = batch.decode("utf-8")
-    data = json.loads(batch)
+        # Try to infer the SampleBatchType (SampleBatch or MultiAgentBatch).
+        if "type" in json_data:
+            data_type = json_data.pop("type")
+        else:
+            raise ValueError("JSON record missing 'type' field")
 
-    if "type" in data:
-        data_type = data.pop("type")
-    else:
-        raise ValueError("JSON record missing 'type' field")
+        if data_type == "SampleBatch":
+            if self.ioctx.worker is not None and \
+                    len(self.ioctx.worker.policy_map) != 1:
+                raise ValueError(
+                    "Found single-agent SampleBatch in input file, but our "
+                    "PolicyMap contains more than 1 policy!")
+            for k, v in json_data.items():
+                json_data[k] = unpack_if_needed(v)
+            if self.ioctx.worker is not None:
+                policy = next(iter(self.ioctx.worker.policy_map.values()))
```

Comment:
this looks a little weird, why iter then next? is this just
self.ioctx.worker.policy_map.values()[0] ?

Reply:
```
>>> d = {"a": 0}
>>> d.values()[0]
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: 'dict_values' object is not subscriptable
```

Reply:
list(values) would force the entirety of values to be placed in a list first, which may not be necessary. We just want to start iterating and stop after the first iter.

Reply:
oh yeah, python3 dict.values() is a dict_value thing.
ok, my bad.